app/src/main/java/com/kickstarter/services/firebase/RegisterService.java

```diff
@@ -18,35 +17,32 @@
 
 import timber.log.Timber;
 
-public class RegisterService extends IntentService {
+public class RegisterService extends JobService {
   protected @Inject ApiClientType apiClient;
   protected @Inject CurrentUserType currentUser;
 
-  public RegisterService() {
-    super("RegisterService");
-  }
-
   @Override
   public void onCreate() {
     super.onCreate();
     ((KSApplication) getApplicationContext()).component().inject(this);
   }
 
+
   @Override
-  protected void onHandleIntent(final @Nullable Intent intent) {
-    Timber.d("onHandleIntent");
+  public boolean onStartJob(final JobParameters job) {
+    FirebaseInstanceId.getInstance().getInstanceId().addOnSuccessListener(instanceIdResult -> {
+      final String newToken = instanceIdResult.getToken();
+      Timber.d("newToken", newToken);
+      sendTokenToApi(newToken);
+      subscribeToGlobalTopic();
+    });
 
-    try {
-      final String token = FirebaseInstanceId.getInstance().getToken();
-      Timber.d("Token: %s", token);
+    return true;
+  }
 
-      if(ObjectUtils.isNotNull(token)) {
-        sendTokenToApi(token);
-        subscribeToGlobalTopic();
-      }
-    } catch (final Exception e) {
-      Timber.e("Failed to complete token refresh: %s", e);
-    }
+  @Override
+  public boolean onStopJob(final JobParameters job) {
+    return false;
```

Comment:
what does this mean and why do we return false?

Reply:
I guess if we needed to kill the job we would make sure we dispose of it here.